File: lib/src/enums/destiny_energy_type.dart
import 'package:json_annotation/json_annotation.dart';

enum DestinyEnergyType {
  ///Any = 0
  @JsonValue(0)
  Any,
  ///Arc = 1
  @JsonValue(1)
  Arc,
  ///Thermal = 2
  @JsonValue(2)
  Thermal,
  ///Void = 3
  @JsonValue(3)
  Void,
}

extension DestinyEnergyTypeExtension on DestinyEnergyType{
  int get value {
    switch(this){
      case DestinyEnergyType.Any:
        return 0;
      case DestinyEnergyType.Arc:
        return 1;
      case DestinyEnergyType.Thermal:
        return 2;
      case DestinyEnergyType.Void:
        return 3;
      default:
        return null;
    }
  }
}

File: BungieNetApi/Models/DestinyEnergyType.cs
using System.Runtime.Serialization;

namespace GhostSharper.Models
{
    /// <summary>
    /// Represents the socket energy types for Armor 2.0, Ghosts 2.0, and Stasis subclasses.
    /// </summary>
    
    public enum DestinyEnergyType
    {
        /// <summary>
        /// Any = 0
        /// </summary>
        [EnumMember(Value = "0")]
        Any = 0,

        /// <summary>
        /// Arc = 1
        /// </summary>
        [EnumMember(Value = "1")]
        Arc = 1,

        /// <summary>
        /// Thermal = 2
        /// </summary>
        [EnumMember(Value = "2")]
        Thermal = 2,

        /// <summary>
        /// Void = 3
        /// </summary>
        [EnumMember(Value = "3")]
        Void = 3,

        /// <summary>
        /// Ghost = 4
        /// </summary>
        [EnumMember(Value = "4")]
        Ghost = 4,

        /// <summary>
        /// Subclass = 5
        /// </summary>
        [EnumMember(Value = "5")]
        Subclass = 5,

        /// <summary>
        /// Stasis = 6
        /// </summary>
        [EnumMember(Value = "6")]
        Stasis = 6,

    }
}